examples/src/java/com/twitter/heron/examples/streamlet/FilesystemSinkTopology.java

```diff
@@ -0,0 +1,141 @@
+//  Copyright 2017 Twitter. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package com.twitter.heron.examples.streamlet;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.logging.Logger;
+
+import com.twitter.heron.api.utils.Utils;
+import com.twitter.heron.examples.streamlet.utils.StreamletUtils;
+import com.twitter.heron.streamlet.Builder;
+import com.twitter.heron.streamlet.Config;
+import com.twitter.heron.streamlet.Context;
+import com.twitter.heron.streamlet.Runner;
+import com.twitter.heron.streamlet.Sink;
+
+/**
+ * This topology demonstrates how sinks work in the Heron Streamlet API for Java.
+ * In this case, the sink is a temporary file. Each value that enters the graph
+ * from the source streamlet (an indefinite series of randomly generated
+ * integers) is written to that temporary file.
+ */
+public final class FilesystemSinkTopology {
+  private FilesystemSinkTopology() {
+  }
+
+  private static final Logger LOG =
+      Logger.getLogger(FilesystemSinkTopology.class.getName());
+
+  /**
+   * Implements the Sink interface, which defines what happens when the toSink
+   * method is invoked in a processing graph.
+   */
+  private static class FilesystemSink<T> implements Sink<T> {
+    private static final long serialVersionUID = -96514621878356224L;
+    private Path tempFilePath;
+    private File tempFile;
+
+    FilesystemSink(File f) {
+      this.tempFile = f;
+    }
+
+    /**
+     * The setup function is called before the sink is used. Any complex
+     * instantiation logic for the sink should go here.
+     */
+    public void setup(Context context) {
+      this.tempFilePath = Paths.get(tempFile.toURI());
+    }
+
+    /**
+     * The put function defines how each incoming streamlet element is
+     * actually processed. In this case, each incoming element is converted
+     * to a byte array and written to the temporary file (successful writes
+     * are also logged). Any exceptions are converted to RuntimeExceptions,
+     * which will effectively kill the topology.
+     */
+    public void put(T element) {
+      byte[] bytes = String.format("%s\n", element.toString()).getBytes();
+
+      try {
+        Files.write(tempFilePath, bytes, StandardOpenOption.APPEND);
```

Comment:
Does Files.write do an open/write/close? Isnt that inefficient to do it per element

Reply:
It probably is less efficient than a lot of other alternatives, but it's only happening every half second and I like that it's just a one liner, which keeps the code clean